test/test.js

```diff
@@ -3050,6 +3050,23 @@ describe('Page', function() {
     });
   });
 
+  describe('Page.setCacheEnabled', function() {
+    it('should enable or disable the cache based on the state passed', async({page, server}) => {
+      let responses = [];
+      page.on('response', response => {
+        if (response.url().includes('.css'))
+          responses.push(response);
+      });
+      await page.setCacheEnabled(false);
+      await page.goto(server.PREFIX + '/cached/one-style.html', {waitUntil: 'networkidle2'});
```

Comment:
two nits:
1. Let's start from making sure it's cached, and then busting caching with this method
2. Can we load-and-reload initially to make sure the stylesheet is cached?

Reply:
I have made these changes now. I needed to reset the `responses` array twice. What do you think?

Reply:
ok, I sent a comment but for some reason it didn't get posted to github. 
Let me re-type it here:

> I needed to reset the responses array twice. What do you think?

You can avoid this by subscribing after the `page.goto`. You can also save a few lines by storing resources in the map:

```js
const responses = new Map();
page.on('response', r => responses.set(r.url().split('/').pop(), r));

await page.goto(server.PREFIX + '/cached/one-style.html');
await page.reload();
expect(responses.get('one-style.css').fromCache()).toBe(true);

await page.setCacheEnabled(false);
await page.reload();
expect(responses.get('one-style.css').fromCache()).toBe(true);
```

Reply:
Ah nice, didn't think of subscribing to the event later. Could be an issue when the responses map gets larger because of no filtering due to more resources being added. But I like this for the current solution.

Reply:
Just a note for anyone else who goes to look at the above code the second expectation should be `toBe(false)`.

Reply:
Have made those changes now.